flang/lib/Parser/prescan.cpp

```diff
@@ -1432,6 +1432,21 @@ Prescanner::IsFixedFormCompilerDirectiveLine(const char *start) const {
     }
     *sp++ = ToLowerCaseLetter(*p);
   }
+  // A fixed form OpenMP conditional compilation sentinel must satisfy the
+  // following criteria, for initial lines:
+  // - Columns 3 through 5 must have only white space or numbers.
+  // - Column 6 must be space or zero.
+  if (column == 3 && sentinel[0] == '$') {
```

Comment:
Is something like `c$1x` handled?
Should we check if it is column 3, 4, 5 as well?

Reply:
Yes, `c$` is not considered a sentinel in `c$1x`.

If the loop above found a conditional compilation sentinel, it will always break on column 3, when it finds either a white space or digit.